03-rnaseq/pathway-analysis_rnaseq_03_gsva.Rmd

```diff
@@ -0,0 +1,657 @@
+
+Let's explore what the output of `gsva()` looks like.
+
+```{r}
+head(gsva_results[, 1:10])
+```
+
+## Write results to file
+
+Let's write all of our GSVA results to file.
+
+```{r}
+gsva_results %>%
+  as.data.frame() %>%
+  tibble::rownames_to_column("pathway") %>%
+  readr::write_tsv(file.path(
+    results_dir,
+    "SRP140558_gsva_results.tsv"
+  ))
+```
+
+**Review**
+
+## Visualizing Results with a heatmap
+
+Let's make a heatmap for our pathways! 
+
+## Neaten up our metadata labels
+
+We will want our heatmap to include our metadata labels, but unfortunately some of the metadata for this dataset are not set up into separate, neat columns. 
+No worries, this is something we will accomplish ourselves by creating a new data frame with the labels we want shown in the heatmap. 
+
+For this study, PBMCs were collected at two time points: during the patients' first, acute bronchiolitis visit (abbreviated "AV") and their recovery visit, (called post-convalescence and abbreviated "CV").
+
+We can create a new variable, `time_point`, that states this info more clearly. 
+This new `time_point` variable will have two labels: `acute illness` and `recovering` based on the `AV` or `CV` coding located in the `refinebio_title` string variable. 
+
+```{r}
+annot_df <- metadata %>%
+  # We need the sample IDs and the main column that contains the metadata info
+  dplyr::select(
+    refinebio_accession_code,
+    refinebio_title
+  ) %>%
+  # Create our `time_point` variable based on `refinebio_title`
+  dplyr::mutate(
+    time_point = dplyr::case_when(
+      # Create our new variable based on refinebio_title containing AV/CV
+      stringr::str_detect(refinebio_title, "_AV_") ~ "acute illness",
+      stringr::str_detect(refinebio_title, "_CV_") ~ "recovering"
+    ),
+    # It's easier for future items if this is already set up as a factor
+    time_point = as.factor(time_point)
+  ) %>%
+  # We don't need the older version of the variable anymore
+  dplyr::select(-refinebio_title)
+```
+
+Let's double check that our factor set up is right. 
+We want `acute illness` to be the first level since it was the first time point collected. 
+
+```{r}
+levels(annot_df$time_point)
+```
+
+The `pheatmap::pheatmap()` will want the annotation data frame to have matching row names to the data we supply it (which is our `gsva_results`).
+
+```{r}
+annot_df <- annot_df %>%
+  # pheatmap will want our sample names that match our data to
+  tibble::column_to_rownames("refinebio_accession_code")
+```
+
+### Set up the heatmap itself
+
+Great! We're all set. 
+We can see that they are in a wide format with the gsva scores for each sample spread across a row associated with each pathway.
+
+```{r}
+heatmap <- pheatmap::pheatmap(gsva_results,
+  annotation_col = annot_df, # Add metadata labels!
+  show_colnames = FALSE, # Don't show sample labels
+  fontsize_row = 6 # Shrink the labels a tad
+)
+
+# Print out heatmap here
+heatmap
```

Comment:
I might add one line about something interesting in the heatmap and also it sounds like (from our 'in person' meeting) there are paired samples here that we also might want to _mention_ if not include as annotation.

Reply:
I tried including the labels for the paired samples, it just looked chaotic. But yeah, they deserve a mention.